framework/src/base/NonlinearSystem.C

```diff
@@ -1786,16 +1795,65 @@ NonlinearSystem::computeJacobianInternal(SparseMatrix<Number> &  jacobian)
       {
         _fe_problem.reinitNodeFace(node, boundary_id, 0);
```

Comment:
I forget - do we know that boundary nodes have at least one boundary condition on them?  We should make sure that we're not calling this when there aren't any BCs on this node...

Note: I know this is the way it was before... just talking about it.

However: we _do_ have to do this before we call `shouldApply()`... so we should be careful...